GVFS/GVFS.PostIndexChangedHook/main.cpp

```diff
@@ -0,0 +1,85 @@
+#include "stdafx.h"
+#include "common.h"
+
+enum PostIndexChangedErrorReturnCode
+{
+    ErrorPostIndexChangedProtocol = ReturnCode::LastError + 1,
+};
+
+const int PIPE_BUFFER_SIZE = 1024;
+
+int main(int argc, char *argv[])
+{
+    if (argc != 3)
+    {
+        die(ReturnCode::InvalidArgCount, "Invalid arguments");
+    }
+
+    if (strcmp(argv[1], "1") && strcmp(argv[1], "0"))
+    {
+        die(PostIndexChangedErrorReturnCode::ErrorPostIndexChangedProtocol, "Invalid value passed for first argument");
+    }
+
+    if (strcmp(argv[2], "1") && strcmp(argv[2], "0"))
+    {
+        die(PostIndexChangedErrorReturnCode::ErrorPostIndexChangedProtocol, "Invalid value passed for second argument");
+    }
+
+    DisableCRLFTranslationOnStdPipes();
+
+    PATH_STRING pipeName(GetGVFSPipeName(argv[0]));
+    PIPE_HANDLE pipeHandle = CreatePipeToGVFS(pipeName);
+
+    // Construct index changed request message
+    // Format:  "PICN|<working directory updated flag><only index updated flag>"

```

Comment:
I think the second flag name needs to be changed.  It used to be the `reset --mixed` flag to indicate that there was a change to the index and it needed to parse the index to update the modified paths instead of the placeholders, but `update-index` was also a command that needs to pass this flag.  The `00` is for commands like `add`, `commit`, etc that update the index and don't need to parse the index but VFSForGit still needs the notification to clear the `newlyCreatedFileAndFolderPaths` list.